corehq/messaging/scheduling/forms.py

```diff
@@ -2566,12 +2555,24 @@ def clean_user_data_property_json(self):
         if not value:
             raise ValidationError(_("This field is required."))
 
+        def json_error():
+            return ValidationError(_("Invalid JSON value: Needs to be and object with valid "
+                                     "properties as keys and strings as values"))
         try:
             value = json.loads(value)
         except Exception:
-            raise ValidationError(_("Invalid JSON value. Expected a list of strings."))
-        # TODO: check that keys are strings and that they are user case fields
-        # Check that values are lists of strings
+            raise json_error()
+        if not isinstance(value, dict):
+            raise json_error()
+
+        if self.cleaned_data.get("use_user_case_for_filter"):
+            from corehq.apps.data_dictionary.util import get_data_dict_props_by_case_type
+            user_case_properties = (
+                get_data_dict_props_by_case_type(self.domain, exclude_deprecated=True))[USERCASE_TYPE]
+            if any(property_name not in user_case_properties for property_name in value.keys()):
+                raise json_error()
```

Comment:
nit: Great to catch this validation, but would that error message be accurate? The error specifically is that the property isn't valid right? In which case, even though it's captured in the generic error message, it would be more helpful to state the specific problem